tests/tests.py

```diff
@@ -0,0 +1,81 @@
+import unittest
```

Comment:
Use pytest or unittest for testing, not both please. We need to keep our tests consistent for maintainability. 

I have a strong preference for using pytest over unittest (if you disagree please let me know and we can discuss).